scipy/stats/_axis_nan_policy.py

```diff
@@ -268,19 +289,19 @@ def _check_empty_inputs(samples, axis):
 masked array with ``mask=False``.""").split('\n')
 
 
-def _axis_nan_policy_factory(result_object, default_axis=0,
+def _axis_nan_policy_factory(tuple_to_result, default_axis=0,
                              n_samples=1, paired=False,
-                             result_unpacker=None, too_small=0,
+                             result_to_tuple=None, too_small=0,
                              n_outputs=2, kwd_samples=[]):
     """Factory for a wrapper that adds axis/nan_policy params to a function.
 
     Parameters
     ----------
-    result_object : callable
+    tuple_to_result : callable
         Callable that returns an object of the type returned by the function
         being wrapped (e.g. the namedtuple or dataclass returned by a
-        statistical test) provided the separate components (e.g. statistic,
-        pvalue).
+        statistical test) provided a tuple of separate components (e.g.
+        statistic, pvalue).
```

Comment:
This should be reverted. Look at the calls to `tuple_to_result` - the individual components need to be provided (but they are always coming from a tuple, which is why this is not an unreasonable name).
```suggestion
        statistical test) provided the separate components (e.g. statistic,
        pvalue).
```
The reason a `tuple_to_result` callable should accept separate arguments rather than a single tuple is that the typical `tuple_to_result` is the initializer of a hypothesis test result class or namedtuple.